cpp/src/arrow/util/tracing_internal.h

```diff
@@ -88,9 +88,10 @@ Iterator<T> WrapIterator(
 
 template <typename T>
 AsyncGenerator<T> WrapAsyncGenerator(AsyncGenerator<T> wrapped,
+                                     opentelemetry::trace::StartSpanOptions options,
                                      const std::string& span_name) {
   return [=]() mutable -> Future<T> {
-    auto span = GetTracer()->StartSpan(span_name);
+    auto span = GetTracer()->StartSpan(span_name, {}, options);
```

Comment:
Can we do `StartSpan(span_name, options)`?  If not, then let's do `StartSpan(span_name, /*attributes=*/{}, options)`

Reply:
I've refactored this a little bit; options has been removed here